### PR TITLE
Added getifaddrs OSX bindings

### PR DESCRIPTION
**Pull request by ferencdg**

### CODE REVIEW

**Review by Geod24 — CHANGES_REQUESTED** [review]

File: src/core/sys/darwin/ifaddrs.d [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    License:    $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+    Authors:    Daniel Graczer
+
+*******************************************************************************/
+
+module core.sys.darwin.ifaddrs;
+
+import core.sys.posix.sys.socket;
+
+version (OSX):
```
Comment (jacob-carlborg) [inline src/core/sys/darwin/ifaddrs.d:21]: The package name contains `darwin` but the version statement says `OSX`. Please add version statements for other Darwin platforms as well, like this: https://github.com/dlang/druntime/blob/17fb8437d4444e212a1a7c79792052abfcda007a/src/core/sys/darwin/dlfcn.d#L12-L21

File: src/core/sys/darwin/ifaddrs.d [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    /// Point-to-point destination addresss
+    sockaddr* if_dstaddr;
+    /// Address specific data
+    void* ifa_data;
+}
+
+/// Returns: linked list of ifaddrs structures describing interfaces
+int getifaddrs(ifaddrs** );
+/// Frees the linked list returned by getifaddrs
+void freeifaddrs(ifaddrs* );
```
Comment (jacob-carlborg) [inline src/core/sys/darwin/ifaddrs.d:48]: What about the `getifmaddrs` and `freeifmaddrs` functions?

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

**Comment by Geod24** [comment]
Ping @ferencdg

**Comment by ferencdg** [comment]
made the suggested changes, thanks for the comments

**Review by jacob-carlborg — APPROVED** [review]

**Comment by thewilsonator** [comment]
```
grep -n -U -P "([ \t]$|\r)" $list_of_files ; test "$?" -ne 0
src/core/sys/darwin/ifaddrs.d:63:    /// Interface	name (AF_LINK)
src/core/sys/darwin/ifaddrs.d:65:    /// Multicast	address
posix.mak:363: recipe for target 'checkwhitespace' failed
```

please remove the trailing whitespace.

**Comment by Geod24** [comment]
> It was actually tabs :|

**Comment by Geod24** [comment]
Fixed the tabs and slightly edited the documentation to refer to the header as opposed to just the (original) functions.

**Comment by Geod24** [comment]
Urgh, this should have been in `COPY` & co as well...

**Pull request merged**